interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java

```diff
@@ -519,30 +525,45 @@ private void runTest(TestCases testCase) throws Exception {
       case RPC_SOAK: {
         tester.performSoakTest(
             serverHost,
-            false /* resetChannelPerIteration */,
             soakIterations,
             soakMaxFailures,
             soakPerIterationMaxAcceptableLatencyMs,
             soakMinTimeMsBetweenRpcs,
             soakOverallTimeoutSeconds,
             soakRequestSize,
-            soakResponseSize);
+            soakResponseSize,
+            numThreads,
+            (currentChannel) -> {
+              try {
+                return tester.maybeCreateNewChannel(currentChannel, false);
+              } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+              }
+            }
+        );
         break;
       }
 
       case CHANNEL_SOAK: {
         tester.performSoakTest(
             serverHost,
-            true /* resetChannelPerIteration */,
             soakIterations,
             soakMaxFailures,
             soakPerIterationMaxAcceptableLatencyMs,
             soakMinTimeMsBetweenRpcs,
             soakOverallTimeoutSeconds,
             soakRequestSize,
-            soakResponseSize);
+            soakResponseSize,
+            numThreads,
+            (currentChannel) -> {
+              try {
+                return tester.maybeCreateNewChannel(currentChannel, true);
```

Comment:
nit: we might just have `maybeCreateNewChannel` itself catch `InterruptedException` and re-throw as `RuntimeException` (so that . That way `maybeCreateNewChannel` no longer throws a checked exception, so we don't need to re-type this try/catch in every lambda we pass to `performSoakTest`